subscriptions-api-openapi.yaml

```diff
@@ -131,120 +138,341 @@ components:
     SubscriptionRequest:
       properties:
         protocol:
-          type: string
-          enum: ["HTTP", "MQTT3", "MQTT5", "AMQP", "NATS", "KAFKA"]
-          description: REQUIRED. Identifier of a delivery protocol.
-          example: "HTTP"
-        protocolsettings:
-          type: object
-          anyOf:
-            - $ref: "#/components/schemas/HTTPSettings"
-            - $ref: "#/components/schemas/MQTTSettings"
-            - $ref: "#/components/schemas/AMQPSettings"
-            - $ref: "#/components/schemas/NATSSettings"
-            - $ref: "#/components/schemas/ApacheKafkaSettings"
-          description: OPTIONAL. A set of settings specific to the selected delivery protocol provider.
+          $ref: "#/components/schemas/Protocol"
+        protocolSettings:
+          oneOf:
+            - $ref: "#/components/schemas/ProtocolSettings"
+            - $ref:  "#/components/schemas/AMQPSettings"
+            - $ref:  "#/components/schemas/ApacheKafkaSettings"
+            - $ref:  "#/components/schemas/HTTPSettings"
+            - $ref:  "#/components/schemas/MQTTSettings"
+            - $ref:  "#/components/schemas/NATSSettings"
         sink:
           type: string
           format: url
           description: REQUIRED. The address to which events shall be delivered using the selected protocol.
           example: "https://endpoint.example.com/webhook"
+        sinkCredentials:
+          oneOf:
+            - $ref: "#/components/schemas/SinkCredential"
+            - $ref: "#/components/schemas/AccessTokenCredential"
+            - $ref: "#/components/schemas/PlainCredential"
+            - $ref: "#/components/schemas/RefreshTokenCredential"
+        source:
+          type: string
+          format: uri-reference
+          description: OPTIONAL. Source to which the subscription applies. May be implied by the request address.
+        types:
+          description: "CloudEvent types eligible to be delivered by this subscription"
+          type: array
+          items:
+            type: string
         filter:
-          $ref: "#/components/schemas/Filter"
+          oneOf:
+            - $ref: "#/components/schemas/Filter"
+            - $ref: "#/components/schemas/AllFilter"
+            - $ref: "#/components/schemas/AnyFilter"
+            - $ref: "#/components/schemas/NotFilter"
+            - $ref: "#/components/schemas/ExactFilter"
+            - $ref: "#/components/schemas/PrefixFilter"
+            - $ref: "#/components/schemas/SuffixFilter"
+            - $ref: "#/components/schemas/DialectFilter"
+        config:
+          description: OPTIONAL. Implementation-specific configuration parameters needed by the subscription manager for acquiring events.
+          type: object
+          additionalProperties:
+            type: string
       required:
+        - id
```

Comment:
I don't think "id" should be on the SubscriptionRequest, but "sink" is required

Reply:
Correcting.